Fix docker error in (local) CI

Traceback
```
E           Exception: Command docker compose -f "/home/tommaso/Fractal/fractal-server/tests/data/slurm_docker_images/docker-compose.yml" -p "pytest112488" up --build -d returned 17: """#0 building with "default" instance using docker driver
E           
E           #1 [slurm-docker-master internal] load build definition from Dockerfile
E           #1 transferring dockerfile: 469B done
E           #1 DONE 0.0s
E           
E           #2 [slurm-docker-master internal] load .dockerignore
E           #2 transferring context: 2B done
E           #2 DONE 0.0s
E           
E           #3 [slurm-docker-master internal] load metadata for docker.io/rancavil/slurm-master:19.05.5-1
E           #3 DONE 0.7s
E           
E           #4 [slurm-docker-master 1/7] FROM docker.io/rancavil/slurm-master:19.05.5-1@sha256:1979f86166b58213380604dcd7c1fcdb2438a40c44add2ff356be47160a97ab3
E           #4 DONE 0.0s
E           
E           #5 [slurm-docker-master internal] load build context
E           #5 transferring context: 59B done
E           #5 DONE 0.0s
E           
E           #6 [slurm-docker-master 4/7] COPY slurm.conf /etc/slurm-llnl/
E           #6 CACHED
E           
E           #7 [slurm-docker-master 2/7] RUN apt-get update
E           #7 CACHED
E           
E           #8 [slurm-docker-master 3/7] RUN sudo apt-get install python3-venv -y
E           #8 CACHED
E           
E           #9 [slurm-docker-master 5/7] ADD sudoers /etc/sudoers
E           #9 CACHED
E           
E           #10 [slurm-docker-master 6/7] RUN useradd --uid 2000 -m fractal -s /usr/bin/bash -d /home/fractal && echo "fractal:fractal" | chpasswd
E           #10 CACHED
E           
E           #11 [slurm-docker-master 7/7] RUN useradd --uid 2001 -m test01 -s /usr/bin/bash -d /home/test01 && echo "test01:test01" | chpasswd
E           #11 CACHED
E           
E           #12 [slurm-docker-master] exporting to image
E           #12 exporting layers done
E           #12 writing image sha256:49d78d0d62ed2f8bfc39c927ddbf32e7944fa69a68dfbc0b439e95534d60a52f done
E           #12 naming to docker.io/library/pytest112488-slurm-docker-master done
E           #12 DONE 0.0s
E           
E           #13 [slurmnode1 internal] load .dockerignore
E           #13 transferring context: 2B done
E           #13 DONE 0.0s
E           
E           #14 [slurmnode1 internal] load build definition from Dockerfile
E           #14 transferring dockerfile: 521B done
E           #14 DONE 0.0s
E           
E           #15 [slurmnode1 internal] load metadata for docker.io/rancavil/slurm-node:19.05.5-1
E           #15 DONE 0.5s
E           
E           #16 [slurmnode1 1/9] FROM docker.io/rancavil/slurm-node:19.05.5-1@sha256:ae650d12fbdaddd29208d7638aa0498c655bfe5a33f4fd07d57e51eb211f18c2
E           #16 DONE 0.0s
E           
E           #17 [slurmnode1 internal] load build context
E           #17 transferring context: 14.92MB 0.1s done
E           #17 DONE 0.1s
E           
E           #18 [slurmnode1 3/9] RUN sudo ln -s /usr/bin/python3.9 /usr/bin/python3
E           #18 CACHED
E           
E           #19 [slurmnode1 2/9] RUN sudo rm -f /usr/bin/python3
E           #19 CACHED
E           
E           #20 [slurmnode1 4/9] RUN sudo apt-get install python3-pip
E           #20 CACHED
E           
E           #21 [slurmnode1 5/9] COPY slurm.conf /etc/slurm-llnl/
E           #21 CACHED
E           
E           #22 [slurmnode1 6/9] ADD fractal_server_local.tar.gz .
E           #22 DONE 0.4s
E           
E           #23 [slurmnode1 7/9] RUN pip3 install ./fractal-server[slurm]
E           #23 0.769 Processing /fractal-server
E           #23 0.986   Installing build dependencies: started
E           #23 1.857   Installing build dependencies: finished with status 'done'
E           #23 1.858   Getting requirements to build wheel: started
E           #23 1.904   Getting requirements to build wheel: finished with status 'done'
E           #23 1.905     Preparing wheel metadata: started
E           #23 2.032     Preparing wheel metadata: finished with status 'done'
E           #23 2.129 Collecting fastapi-users[oauth]<13.0.0,>=12.1.0
E           #23 2.153   Downloading fastapi_users-12.1.2-py3-none-any.whl (38 kB)
E           #23 2.649 Collecting sqlalchemy[asyncio]<2.0,>=1.4
E           #23 2.664   Downloading SQLAlchemy-1.4.50-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.6 MB)
E           #23 3.042 Collecting pydantic<2,>=1.10.8
E           #23 3.051   Downloading pydantic-1.10.13-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.2 MB)
E           #23 3.335 Collecting fastapi<0.104.0,>=0.103.0
E           #23 3.343   Downloading fastapi-0.103.2-py3-none-any.whl (66 kB)
E           #23 3.385 Collecting python-dotenv<0.21.0,>=0.20.0
E           #23 3.390   Downloading python_dotenv-0.20.0-py3-none-any.whl (17 kB)
E           #23 3.434 Collecting uvicorn<0.21.0,>=0.20.0
E           #23 3.440   Downloading uvicorn-0.20.0-py3-none-any.whl (56 kB)
E           #23 3.470 Collecting sqlmodel<0.0.12,>=0.0.11
E           #23 3.517   Downloading sqlmodel-0.0.11-py3-none-any.whl (21 kB)
E           #23 3.554 Collecting aiosqlite<0.18.0,>=0.17.0
E           #23 3.559   Downloading aiosqlite-0.17.0-py3-none-any.whl (15 kB)
E           #23 3.606 Collecting alembic<2.0.0,>=1.9.1
E           #23 3.613   Downloading alembic-1.12.1-py3-none-any.whl (226 kB)
E           #23 3.659 Collecting cloudpickle<2.3.0,>=2.2.1; extra == "slurm"
E           #23 3.666   Downloading cloudpickle-2.2.1-py3-none-any.whl (25 kB)
E           #23 3.691 Collecting clusterfutures<0.6,>=0.5; extra == "slurm"
E           #23 3.697   Downloading clusterfutures-0.5-py3-none-any.whl (9.6 kB)
E           #23 3.728 Collecting pyjwt[crypto]==2.8.0
E           #23 3.734   Downloading PyJWT-2.8.0-py3-none-any.whl (22 kB)
E           #23 3.767 Collecting makefun<2.0.0,>=1.11.2
E           #23 3.783   Downloading makefun-1.15.1-py2.py3-none-any.whl (22 kB)
E           #23 3.813 Collecting email-validator<2.1,>=1.1.0
E           #23 3.820   Downloading email_validator-2.0.0.post2-py3-none-any.whl (31 kB)
E           #23 3.849 Collecting python-multipart==0.0.6
E           #23 3.855   Downloading python_multipart-0.0.6-py3-none-any.whl (45 kB)
E           #23 3.903 Collecting passlib[bcrypt]==1.7.4
E           #23 3.910   Downloading passlib-1.7.4-py2.py3-none-any.whl (525 kB)
E           #23 3.975 Collecting httpx-oauth>=0.13; extra == "oauth"
E           #23 3.981   Downloading httpx_oauth-0.13.0-py3-none-any.whl (30 kB)
E           #23 4.250 Collecting greenlet!=0.4.17; python_version >= "3" and (platform_machine == "aarch64" or (platform_machine == "ppc64le" or (platform_machine == "x86_64" or (platform_machine == "amd64" or (platform_machine == "AMD64" or (platform_machine == "win32" or platform_machine == "WIN32"))))))
E           #23 4.258   Downloading greenlet-3.0.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (657 kB)
E           #23 4.328 Collecting typing-extensions>=4.2.0
E           #23 4.337   Downloading typing_extensions-4.8.0-py3-none-any.whl (31 kB)
E           #23 4.367 Collecting anyio<4.0.0,>=3.7.1
E           #23 4.373   Downloading anyio-3.7.1-py3-none-any.whl (80 kB)
E           #23 4.425 Collecting starlette<0.28.0,>=0.27.0
E           #23 4.431   Downloading starlette-0.27.0-py3-none-any.whl (66 kB)
E           #23 4.470 Collecting h11>=0.8
E           #23 4.480   Downloading h11-0.14.0-py3-none-any.whl (58 kB)
E           #23 4.515 Collecting click>=7.0
E           #23 4.520   Downloading click-8.1.7-py3-none-any.whl (97 kB)
E           #23 4.561 Collecting sqlalchemy2-stubs
E           #23 4.566   Downloading sqlalchemy2_stubs-0.0.2a36-py3-none-any.whl (191 kB)
E           #23 4.621 Collecting Mako
E           #23 4.632   Downloading Mako-1.2.4-py3-none-any.whl (78 kB)
E           #23 4.904 Collecting cryptography>=3.4.0; extra == "crypto"
E           #23 4.911   Downloading cryptography-41.0.5-cp37-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.4 MB)
E           #23 5.163 Collecting idna>=2.0.0
E           #23 5.169   Downloading idna-3.4-py3-none-any.whl (61 kB)
E           #23 5.211 Collecting dnspython>=2.0.0
E           #23 5.219   Downloading dnspython-2.4.2-py3-none-any.whl (300 kB)
E           #23 5.317 Collecting bcrypt>=3.1.0; extra == "bcrypt"
E           #23 5.324   Downloading bcrypt-4.0.1-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (593 kB)
E           #23 5.396 Collecting httpx<0.25,>=0.18
E           #23 5.402   Downloading httpx-0.24.1-py3-none-any.whl (75 kB)
E           #23 5.439 Collecting exceptiongroup; python_version < "3.11"
E           #23 5.444   Downloading exceptiongroup-1.1.3-py3-none-any.whl (14 kB)
E           #23 5.467 Collecting sniffio>=1.1
E           #23 5.472   Downloading sniffio-1.3.0-py3-none-any.whl (10 kB)
E           #23 5.550 Collecting MarkupSafe>=0.9.2
E           #23 5.556   Downloading MarkupSafe-2.1.3-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
E           #23 5.741 Collecting cffi>=1.12
E           #23 5.751   Downloading cffi-1.16.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (443 kB)
E           #23 5.807 Collecting httpcore<0.18.0,>=0.15.0
E           #23 5.814   Downloading httpcore-0.17.3-py3-none-any.whl (74 kB)
E           #23 5.859 Collecting certifi
E           #23 5.865   Downloading certifi-2023.7.22-py3-none-any.whl (158 kB)
E           #23 5.897 Collecting pycparser
E           #23 5.901   Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)
E           #23 5.913 Building wheels for collected packages: fractal-server
E           #23 5.913   Building wheel for fractal-server (PEP 517): started
E           #23 6.090   Building wheel for fractal-server (PEP 517): finished with status 'error'
E           #23 6.090   ERROR: Command errored out with exit status 1:
E           #23 6.090    command: /usr/bin/python3 /tmp/tmp5a0f4ywj_in_process.py build_wheel /tmp/tmp9k9bdddy
E           #23 6.090        cwd: /tmp/pip-req-build-mu51al_a
E           #23 6.090   Complete output (23 lines):
E           #23 6.090   fatal: not a git repository (or any of the parent directories): .git
E           #23 6.090   Traceback (most recent call last):
E           #23 6.090     File "/tmp/tmp5a0f4ywj_in_process.py", line 280, in <module>
E           #23 6.090       main()
E           #23 6.090     File "/tmp/tmp5a0f4ywj_in_process.py", line 263, in main
E           #23 6.090       json_out['return_val'] = hook(**hook_input['kwargs'])
E           #23 6.090     File "/tmp/tmp5a0f4ywj_in_process.py", line 204, in build_wheel
E           #23 6.090       return _build_backend().build_wheel(wheel_directory, config_settings,
E           #23 6.090     File "/tmp/pip-build-env-7e953ibr/overlay/lib/python3.9/site-packages/poetry/core/masonry/api.py", line 58, in build_wheel
E           #23 6.090       return WheelBuilder.make_in(
E           #23 6.090     File "/tmp/pip-build-env-7e953ibr/overlay/lib/python3.9/site-packages/poetry/core/masonry/builders/wheel.py", line 88, in make_in
E           #23 6.090       wb.build(target_dir=directory)
E           #23 6.090     File "/tmp/pip-build-env-7e953ibr/overlay/lib/python3.9/site-packages/poetry/core/masonry/builders/wheel.py", line 124, in build
E           #23 6.090       self._copy_module(zip_file)
E           #23 6.090     File "/tmp/pip-build-env-7e953ibr/overlay/lib/python3.9/site-packages/poetry/core/masonry/builders/wheel.py", line 269, in _copy_module
E           #23 6.090       to_add = self.find_files_to_add()
E           #23 6.090     File "/tmp/pip-build-env-7e953ibr/overlay/lib/python3.9/site-packages/poetry/core/masonry/builders/builder.py", line 206, in find_files_to_add
E           #23 6.090       include_file.relative_to_project_root()
E           #23 6.090     File "/tmp/pip-build-env-7e953ibr/overlay/lib/python3.9/site-packages/poetry/core/masonry/builders/builder.py", line 400, in relative_to_project_root
E           #23 6.090       return self.path.relative_to(self.project_root)
E           #23 6.090     File "/usr/lib/python3.9/pathlib.py", line 928, in relative_to
E           #23 6.090       raise ValueError("{!r} is not in the subpath of {!r}"
E           #23 6.090   ValueError: '/usr/bin/python3.9' is not in the subpath of '/tmp/pip-req-build-mu51al_a' OR one path is relative and the other is absolute.
E           #23 6.090   ----------------------------------------
E           #23 6.090   ERROR: Failed building wheel for fractal-server
E           #23 6.090 Failed to build fractal-server
E           #23 6.157 ERROR: Could not build wheels for fractal-server which use PEP 517 and cannot be installed directly
E           #23 ERROR: process "/bin/sh -c pip3 install ./fractal-server[slurm]" did not complete successfully: exit code: 1
E           ------
E            > [slurmnode1 7/9] RUN pip3 install ./fractal-server[slurm]:
E           6.090       include_file.relative_to_project_root()
E           6.090     File "/tmp/pip-build-env-7e953ibr/overlay/lib/python3.9/site-packages/poetry/core/masonry/builders/builder.py", line 400, in relative_to_project_root
E           6.090       return self.path.relative_to(self.project_root)
E           6.090     File "/usr/lib/python3.9/pathlib.py", line 928, in relative_to
E           6.090       raise ValueError("{!r} is not in the subpath of {!r}"
E           6.090   ValueError: '/usr/bin/python3.9' is not in the subpath of '/tmp/pip-req-build-mu51al_a' OR one path is relative and the other is absolute.
E           6.090   ----------------------------------------
E           6.090   ERROR: Failed building wheel for fractal-server
E           6.090 Failed to build fractal-server
E           6.157 ERROR: Could not build wheels for fractal-server which use PEP 517 and cannot be installed directly
E           ------
E           failed to solve: process "/bin/sh -c pip3 install ./fractal-server[slurm]" did not complete successfully: exit code: 1
E           """.

../../.cache/pypoetry/virtualenvs/fractal-server-cyevISt_-py3.10/lib/python3.10/site-packages/pytest_docker/plugin.py:24: Exception
``

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,8 @@
 
 * Improve delete-resource endpoint (\#943).
 * Expose JSON Schema for the `ManifestV1` Pydantic model (\#942).
+* Testing:
+    * Fix bug in local tests with Docker/SLURM (\#948).
 
 # 1.3.13
 

--- a/tests/data/slurm_docker_images/node/Dockerfile
+++ b/tests/data/slurm_docker_images/node/Dockerfile
@@ -1,6 +1,5 @@
 FROM rancavil/slurm-node:19.05.5-1
 
-
 # Relink python3 (from 3.8 to 3.9)
 RUN sudo rm -f /usr/bin/python3
 RUN sudo ln -s /usr/bin/python3.9 /usr/bin/python3
@@ -8,7 +7,10 @@ RUN sudo apt-get install python3-pip
 
 COPY slurm.conf /etc/slurm-llnl/
 ADD fractal_server_local.tar.gz .
-RUN pip3 install ./fractal-server[slurm]
+WORKDIR ./fractal-server
+
+RUN pip3 install --upgrade pip setuptools wheel
+RUN pip3 install fractal_server[slurm]
 RUN pip3 install devtools
 
 RUN useradd --uid 2001 -m test01 -s /usr/bin/bash -d /home/test01 && echo "test01:test01" | chpasswd

--- a/tests/data/slurm_docker_images/node/Dockerfile
+++ b/tests/data/slurm_docker_images/node/Dockerfile
@@ -10,7 +10,7 @@ ADD fractal_server_local.tar.gz .
 WORKDIR ./fractal-server
 
 RUN pip3 install --upgrade pip setuptools wheel
-RUN pip3 install fractal_server[slurm]
+RUN pip3 install -e .[slurm]
 RUN pip3 install devtools
 
 RUN useradd --uid 2001 -m test01 -s /usr/bin/bash -d /home/test01 && echo "test01:test01" | chpasswd

